joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/utils/ExternalCommand.scala

```diff
@@ -1,30 +1,26 @@
 package io.joern.c2cpg.utils
 
-import scala.collection.mutable
+import java.util.concurrent.ConcurrentLinkedQueue
 import scala.sys.process.{Process, ProcessLogger}
 import scala.util.{Failure, Success, Try}
+import scala.jdk.CollectionConverters._
 
 object ExternalCommand {
 
-  private val IS_WIN: Boolean =
-    scala.util.Properties.isWin
-
-  private val shellPrefix: Seq[String] =
-    if (IS_WIN) "cmd" :: "/c" :: Nil else "sh" :: "-c" :: Nil
+  private val IS_WIN: Boolean = scala.util.Properties.isWin
 
   def run(command: String): Try[Seq[String]] = {
-    val result                      = mutable.ArrayBuffer.empty[String]
-    val lineHandler: String => Unit = result.addOne
-    Process(shellPrefix :+ command).!(ProcessLogger(lineHandler, lineHandler)) match {
-      case 0 =>
-        Success(result.toSeq)
-      case 1
-          if IS_WIN &&
-            command != IncludeAutoDiscovery.GCC_VERSION_COMMAND &&
-            IncludeAutoDiscovery.gccAvailable() =>
-        Success(result.toSeq)
+    val stdOutOutput  = new ConcurrentLinkedQueue[String]
+    val processLogger = ProcessLogger(stdOutOutput.add, stdOutOutput.add)
+    Try(Process(command).!(processLogger)) match {
+      case Success(0) =>
+        Success(stdOutOutput.asScala.toSeq)
+      case Success(1)
+          if IS_WIN && command != IncludeAutoDiscovery.GCC_VERSION_COMMAND && IncludeAutoDiscovery.gccAvailable() =>
+        Success(stdOutOutput.asScala.toSeq)
```

Comment:
I know you didn't change this, but it might be worth adding a comment if one of you knows: 
Looks like there are cases on windows where the external commands returns `1` and that's a success? My understanding so far was that also for windows, anything other than `0` is a failure..?

Reply:
The command to query the system header file locations within a Windows environment always returns Success(1) for whatever reason. I was not able to figure out why that is a thing.

Reply:
👍🏻 thanks! For future readers we could add that as a comment (suggested below)..

Reply:
Also note that future readers is most likely one of us 😆